### PR TITLE
Support OwnerLoginName in New-DbaAgentSchedule

### PR DESCRIPTION
**Pull request by david-garcia-garcia**
## Type of Change

 - [x] New feature (non-breaking change )
### Purpose
It is not possible to create a schedule inside a container when interacting with MSSQL using Integrated Authentication and the user is the default ContainerAdmin:

```
C:\entrypoint\refreshenv\ConfigureMssqlJobs.ps1 : The specified
mssql-1  | '@owner_login_name' is invalid (valid values are returned by sp_helplogins
mssql-1  | [excluding Windows NT groups]).
```

It looks like this default account has some sort of trait that prevents it from being the owner of a Schedule. You can create jobs using other users as owners.

### Approach
Allow to manually set the Ownership of the schedule during creation.

### CODE REVIEW

File: public/New-DbaAgentSchedule.ps1 [hunk 1]
```diff
@@ -99,6 +99,9 @@ function New-DbaAgentSchedule {
 
         If force is used the start time will be '23:59:59'
 
+    .PARAMETER OwnerLoginName
```
Comment (andreasjordan) [inline public/New-DbaAgentSchedule.ps1:102]: Most Owner parameter are named "Owner":

image

Only Set-DbaAgentJob uses OwnerLogin, none uses OwnerLoginName. I would vote for "Owner" as the parameter name.

And you need to add the new parameter also to the test file.

File: public/New-DbaAgentSchedule.ps1 [hunk 2]
```diff
@@ -523,6 +527,10 @@ function New-DbaAgentSchedule {
                         $jobschedule.ActiveEndTimeOfDay = $EndTime
                     }
 
+                    if ($OwnerLoginName) {
+                        $jobschedule.OwnerLoginName = $OwnerLoginName;
```
Comment (andreasjordan) [inline public/New-DbaAgentSchedule.ps1:531]: We do not use ";" at the end of lines.

**Review by andreasjordan — CHANGES_REQUESTED** [review]
Looks good, but I have not tested the code.

File: public/New-DbaAgentSchedule.ps1 [hunk 3]
```diff
@@ -99,6 +99,9 @@ function New-DbaAgentSchedule {
 
         If force is used the start time will be '23:59:59'
 
+    .PARAMETER OwnerLoginName
+        The name of the login that owns the job. login is sysname, with a default of NULL, which is interpreted as the current login name.
```
Comment (wsmelton) [inline public/New-DbaAgentSchedule.ps1:103]:
```suggestion
        Login to own the job, defaults to login running the command.
```

**Review by wsmelton — CHANGES_REQUESTED** [review]
Shorten description a bit to be more to the point

File: public/New-DbaAgentSchedule.ps1 (hunk 3 above)
Comment (andreasjordan) [reply public/New-DbaAgentSchedule.ps1:103]: I like that.

File: public/New-DbaAgentSchedule.ps1 (hunk 1 above)
Comment (david-garcia-garcia) [reply public/New-DbaAgentSchedule.ps1]: @andreasjordan 

> And you need to add the new parameter also to the test file. 

There are several test that directly or indirectly test New-DbaAgentSchedule, do you mean adding specific test coverage for the Owner parameter, or just populating it in one of the existing tests so it gets indirectly tested?

Comment (andreasjordan) [reply public/New-DbaAgentSchedule.ps1]: You need to add the new parameter to this list (between EndTime and Force) - or this test will fail: 
https://github.com/dataplat/dbatools/blob/78c1bc392985541999949f76d2738b316e41a534/tests/New-DbaAgentSchedule.Tests.ps1#L9

You could also add a new test that uses the parameter. But I think that is not needed.

After you have changed the test file, I will activate the tests for your PR.

**Pull request merged**